src/components/common/Calculator.js

```diff
@@ -195,16 +298,83 @@ class Calculator extends Component {
     return '';
   };
 
-  backspace = (result) => {
-    const lastChar = result.slice(-1);
-    let newStr = result.slice(0, -1);
-    // if removed char is part of a group
-    // remove one more visible character
-    const reg = RegExp('[\\{\\}A-Za-z]', 'g');
-    if (reg.exec(lastChar)?.[0]) {
-      newStr = this.backspace(newStr);
+  backSpace = (katexStr, mathjsStr, history) => {
+    const lastOperation = history.pop();
+
+    let newKatex = katexStr;
+    let newMathjs = mathjsStr;
+
+    // if lastOperation does not exist, remove a char
+    if (!lastOperation) {
+      newKatex = newKatex.slice(0, -1);
+      newMathjs = newMathjs.slice(0, -1);
+    } else {
+      const { text, katex: katexExp, mathjs: mathjsExp } = KEYPAD_BUTTONS.find(
+        ({ name }) => name === lastOperation,
+      );
+
+      // find last instance of added operation and remove it
+      const katexReg = new RegExp(parse(`${katexExp || text}$`), 'g').exec(
+        newKatex,
+      );
+      const mathjsReg = new RegExp(parse(`${mathjsExp || text}$`), 'g').exec(
+        newMathjs,
+      );
+      if (katexReg && mathjsReg) {
+        newKatex = newKatex.slice(0, katexReg.index);
+        newMathjs = newMathjs.slice(0, mathjsReg.index);
+      }
     }
-    return newStr;
+
+    return [newKatex, newMathjs];
+  };
+
+  renderFooter = () => {
+    const { scientificMode, angleUnit } = this.state;
+    return (
+      <Grid container justify="center" alignItems="center">
+        <Grid container xs={7}>
+          {this.renderFocusIndicator()}
+        </Grid>
+        <Grid container xs={5} align="right">
+          {this.renderScientificSwitch()}
+          {scientificMode && (
+            <AngleUnitSwitch
+              onChange={this.onChangeAngleUnit}
+              angleUnit={angleUnit}
+            />
+          )}
+        </Grid>
+      </Grid>
+    );
+  };
+
+  onChangeAngleUnit = () => {
+    const { angleUnit } = this.state;
+    this.setState({
+      angleUnit:
+        angleUnit === ANGLE_UNITS.DEG ? ANGLE_UNITS.RAD : ANGLE_UNITS.DEG,
```

Comment:
Can use `||`, no?

Reply:
I am not sure? Or don't see it, because `angleUnit` is a string, so I need it to be either `DEG` or `RAD`.